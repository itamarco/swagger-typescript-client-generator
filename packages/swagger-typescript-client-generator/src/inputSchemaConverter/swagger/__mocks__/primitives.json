{
  "swagger": "2.0",
  "paths": {
    "/primitives/string/{parameter}": {
      "get": {
        "summary": "Gets a random string",
        "parameters": [
          {
            "name": "parameter",
            "in": "path",
            "type": "string",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/StringResponse"
            }
          }
        }
      }
    },
    "/primitives/number/{parameter}": {
      "get": {
        "summary": "Gets a random number",
        "parameters": [
          {
            "name": "parameter",
            "in": "path",
            "type": "number",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/NumberResponse"
            }
          }
        }
      }
    },
    "/primitives/boolean/{parameter}": {
      "get": {
        "summary": "Gets a random boolean",
        "parameters": [
          {
            "name": "parameter",
            "in": "path",
            "type": "boolean",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/BooleanResponse"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "StringResponse": {
      "type": "string",
      "default": "default value",
      "format": "DateTime",
      "description": "string response definition"
    },
    "NumberResponse": {
      "type": "number",
      "format": "float",
      "minimum": 0,
      "maximum": 9,
      "default": "5",
      "description": "number response definition"
    },
    "BooleanResponse": {
      "type": "boolean",
      "description": "boolean response definition"
    }
  }
}
