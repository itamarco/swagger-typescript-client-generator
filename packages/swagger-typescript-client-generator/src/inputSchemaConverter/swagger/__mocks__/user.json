{
  "swagger": "2.0",
  "paths": {
    "/users/{id}": {
      "get": {
        "summary": "Gets a user by ID.",
        "description": "A detailed description of the operation. GitHub Flavored Markdown can be used for rich text representation, such as **bold**, *italic* and [links](https://swagger.io).\n",
        "operationId": "getUsers",
        "tags": ["users"],
        "produces": ["application/json", "application/xml"],
        "parameters": [
          {
            "name": "id",
            "in": "path",
            "description": "User ID",
            "type": "integer",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/User"
            }
          }
        },
        "externalDocs": {
          "url": "http://api.example.com/docs/user-operations/",
          "description": "Learn more about User operations provided by this API."
        }
      }
    }
  },
  "definitions": {
    "User": {
      "type": "object",
      "properties": {
        "id": {
          "type": "integer"
        },
        "name": {
          "type": "string"
        }
      },
      "required": ["id", "name"]
    }
  }
}
