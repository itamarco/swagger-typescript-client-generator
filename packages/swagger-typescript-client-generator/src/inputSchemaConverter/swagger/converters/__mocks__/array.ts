import * as Swagger from "swagger-schema-official"

export const array: Swagger.Schema[] = [
  {
    type: "array",
    description: "type without items"
  },
  {
    type: "array",
    items: {},
    description: "type with single item"
  }
]
