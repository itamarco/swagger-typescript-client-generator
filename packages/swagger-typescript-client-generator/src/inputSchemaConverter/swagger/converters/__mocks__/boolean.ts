import * as Swagger from "swagger-schema-official"

export const boolean: Swagger.Schema[] = [
  {
    type: "boolean",
    description: "type number"
  },
  {
    type: "boolean",
    default: true,
    description: "type and positive default"
  },
  {
    type: "boolean",
    default: false,
    description: "type and negative default"
  },
  {
    type: "boolean",
    default: 0,
    description: "type and numeric negative falsy default"
  },
  {
    type: "boolean",
    default: 1,
    description: "type and numeric truthy default"
  },
  {
    type: "boolean",
    default: "on" as any,
    description: "type and string default"
  },
  {
    type: "boolean",
    format: "timestamp",
    description: "type and format"
  },
  {
    type: "boolean",
    enum: [],
    description: "type and empty enum"
  },
  {
    type: "boolean",
    enum: [false, true],
    description: "type and boolean enum"
  },
  {
    type: "boolean",
    enum: [true, false, "2", 3],
    description: "type and mixed enum"
  }
]
