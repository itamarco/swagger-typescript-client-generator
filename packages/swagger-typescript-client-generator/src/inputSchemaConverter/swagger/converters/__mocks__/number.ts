import * as Swagger from "swagger-schema-official"

export const number: Swagger.Schema[] = [
  {
    type: "number",
    description: "type number"
  },
  {
    type: "integer",
    description: "type integer"
  },
  {
    type: "number",
    minimum: 1,
    description: "type and numeric minimum"
  },
  {
    type: "number",
    minimum: "1" as any,
    description: "type and string minimum"
  },
  {
    type: "number",
    maximum: 1,
    description: "type and numeric maximum"
  },
  {
    type: "number",
    maximum: "1" as any,
    description: "type and string maximum"
  },
  {
    type: "number",
    default: 1,
    description: "type and numeric default"
  },
  {
    type: "number",
    default: "1" as any,
    description: "type and string default"
  },
  {
    type: "number",
    format: "timestamp",
    description: "type and format"
  },
  {
    type: "number",
    enum: [],
    description: "type and empty enum"
  },
  {
    type: "number",
    enum: [1, 2, 3],
    description: "type and numeric enum"
  },
  {
    type: "number",
    enum: [true, "2", 3],
    description: "type and mixed enum"
  }
]
