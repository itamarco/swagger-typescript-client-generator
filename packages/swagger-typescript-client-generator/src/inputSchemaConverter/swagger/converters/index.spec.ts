import { convertSwaggerSchemaToNormalizedSchema } from "./index"
import * as mocks from "./__mocks__"

describe.each(Object.entries(mocks))(
  `swagger definition converter for type "%s"`,
  (type, list) => {
    test.each(list.map(item => [item.description, item]))(
      `converter converts scenario "%s"`,
      (description, definition) => {
        const spec = convertSwaggerSchemaToNormalizedSchema(definition as any)
        expect(spec.type).toEqual(type)
        expect(spec).toMatchSnapshot()
      }
    )
  }
)
