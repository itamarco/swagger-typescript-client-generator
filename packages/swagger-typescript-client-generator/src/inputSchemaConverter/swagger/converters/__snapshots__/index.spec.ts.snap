// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`definition converter convert to "number" schema definition "type and empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and empty enum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "format": "timestamp",
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and numeric default" 1`] = `
Object {
  "default": 1,
  "description": "type and numeric default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and numeric enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and numeric maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": 1,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and numeric minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": 1,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and string default" 1`] = `
Object {
  "default": 1,
  "description": "type and string default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and string maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": "1",
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type and string minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": "1",
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type integer" 1`] = `
Object {
  "default": undefined,
  "description": "type integer",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "number" schema definition "type number" 1`] = `
Object {
  "default": undefined,
  "description": "type number",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`definition converter convert to "string" schema definition "additional fields used" 1`] = `
Object {
  "default": undefined,
  "description": "additional fields used",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "all fields set" 1`] = `
Object {
  "default": undefined,
  "description": "all fields set",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "just type and description" 1`] = `
Object {
  "default": undefined,
  "description": "just type and description",
  "enum": undefined,
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type and defined empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and defined empty enum",
  "enum": undefined,
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    "true",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type and string enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string enum",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type with default value" 1`] = `
Object {
  "default": "default value",
  "description": "type with default value",
  "enum": undefined,
  "type": "string",
}
`;

exports[`definition converter convert to "string" schema definition "type, format and default value" 1`] = `
Object {
  "default": "some default value",
  "description": "type, format and default value",
  "enum": undefined,
  "type": "string",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and boolean enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and boolean enum",
  "enum": Array [
    false,
    true,
  ],
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and empty enum",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    true,
    false,
    false,
    false,
  ],
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and negative default" 1`] = `
Object {
  "default": false,
  "description": "type and negative default",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and numeric negative falsy default" 1`] = `
Object {
  "default": false,
  "description": "type and numeric negative falsy default",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and numeric truthy default" 1`] = `
Object {
  "default": true,
  "description": "type and numeric truthy default",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and positive default" 1`] = `
Object {
  "default": true,
  "description": "type and positive default",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type and string default" 1`] = `
Object {
  "default": true,
  "description": "type and string default",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "boolean" converter converts scenario "type number" 1`] = `
Object {
  "default": undefined,
  "description": "type number",
  "enum": undefined,
  "type": "boolean",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and empty enum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "format": "timestamp",
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and numeric default" 1`] = `
Object {
  "default": 1,
  "description": "type and numeric default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and numeric enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and numeric maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": 1,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and numeric minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": 1,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and string default" 1`] = `
Object {
  "default": 1,
  "description": "type and string default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and string maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": "1",
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type and string minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": "1",
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type integer" 1`] = `
Object {
  "default": undefined,
  "description": "type integer",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "number" converter converts scenario "type number" 1`] = `
Object {
  "default": undefined,
  "description": "type number",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "additional fields used" 1`] = `
Object {
  "default": undefined,
  "description": "additional fields used",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "all fields set" 1`] = `
Object {
  "default": undefined,
  "description": "all fields set",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "just type and description" 1`] = `
Object {
  "default": undefined,
  "description": "just type and description",
  "enum": undefined,
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type and defined empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and defined empty enum",
  "enum": undefined,
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    "true",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type and string enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string enum",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type with default value" 1`] = `
Object {
  "default": "default value",
  "description": "type with default value",
  "enum": undefined,
  "type": "string",
}
`;

exports[`swagger definition converter for type "string" converter converts scenario "type, format and default value" 1`] = `
Object {
  "default": "some default value",
  "description": "type, format and default value",
  "enum": undefined,
  "type": "string",
}
`;
